packages/config/src/ProfileInfo.ts

```diff
@@ -863,11 +865,57 @@ export class ProfileInfo {
      *      into mergedArgs.knownArgs if a value is found in an environment
      *      variable for any missingArgs.
      */
-    private overrideWithEnv(mergedArgs: IProfMergedArg) {
-        if (this.mOverrideWithEnv === false) {
-            return;
-        }
+    private overrideWithEnv(mergedArgs: IProfMergedArg, profSchema?: IProfileSchema) {
+        // populate any missing options
+        const envPrefix = ImperativeConfig.instance.loadedConfig.envVariablePrefix;
+        const envStart = envPrefix + "_OPT_";
+        for (const key in process.env) {
+            if (key.startsWith(envStart)) {
+                let argValue: any = process.env[key];
+                let dataType: any = typeof argValue;
+                const argName: string = CliUtils.getOptionFormat(key.substring(envStart.length).replace(/_/g, "-").toLowerCase()).camelCase;
+
+                let argNameFound = false;
+                if (profSchema != null) {
+                    for (const [propName, propObj] of Object.entries(profSchema.properties || {})) {
+                        if (argName === propName) {
+                            dataType = this.argDataType(propObj.type);
+                            argNameFound = true;
+                        }
+                    }
+                }
 
-        // todo: get values from environment variables
+                if (profSchema == null || !argNameFound) {
+                    if (argValue.toUpperCase() === "TRUE" || argValue.toUpperCase() === "FALSE") {
+                        dataType = "boolean";
+                    } else if (!isNaN(+(argValue))) {
+                        dataType = "number";
+                    }
+                    // TODO: Look for option definition for argName to check if it's an array
+                }
+
+                if (dataType === "boolean") {
+                    argValue = argValue.toUpperCase() === "TRUE";
+                } else if (dataType === "number") {
+                    argValue = +(argValue);
+                } else if (dataType === "array") {
+                    argValue = CliUtils.extractArrayFromEnvValue(argValue);
+                }
+
+                const tempArg: IProfArgAttrs = {
+                    argName,
+                    argValue,
+                    dataType,
+                    argLoc: { locType: ProfLocType.ENV }
+                };
+
+                const argIndex = mergedArgs.knownArgs.findIndex((arg) => arg.argName === argName)
+                if (argIndex < 0) {
+                    mergedArgs.knownArgs.push(tempArg);
+                } else if (this.mOverrideWithEnv) {
```

Comment:
Is there a reason for placing the test of mOverrideWithEnv at the end of this function? If we tested mOverrideWithEnv for FALSE at the beginning of the function, we could immediately return and not perform any of the environment operations.

What have I overlooked?

Reply:
I guess I made the assumption (probably a wrong one :yum:) that we may want to still load environment variables into `knownArgs` in case they were not specified by the user, or don't exist in a profile.

Reply:
The mOverrideWithEnv variable was created because at least some of the VSCode extension teams said that they never want environment variables to override the values from the profiles.  Such teams will specify such a choice on the ProfileInfo constructor, and we will not use the environment to override any argument value for that instance of the ProfileInfo class.

Reply:
My interpretation is that those GUI teams don't want the environment to supply a value even for an argument that is not specified anywhere. They feel that environment variables will be too confusing for  their VSCode users.

Reply:
Thanks Gene. I think I've address these comments. 😋